api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "goals", "verbose" })
+@EqualsAndHashCode
+public class KafkaClusterRebalanceSpec implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private List<String> goals;
+    private boolean verbose = false;
+
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("A list of goals to use in generating the proposal and executing it. " +
+            "The supported goals are available at https://github.com/linkedin/cruise-control#goals." +
+            "Providing empty goals means to use all of them.")
```

Comment:
> Providing empty goals means to use all of them.

All of them in what order (I assume order matters?)

Reply:
as far as I understood, the goals are in decreasing priority order as they are provided to the CC REST API as parameter. If you don't provide goals (so empty as I mentioned in the description), then CC uses the goals specified in the `default.goals` parameter in the configuration properties file. In that case, it's the same, the priority is the order as they are listed there.
How does it sound to you ...

"A list of goals, ordered by decreasing priority, to use in generating the proposal and executing it. " +
"The supported goals are available at https://github.com/linkedin/cruise-control#goals." +
"Providing empty goals means to use all the ones declared in the default.goals configuration parameter"

Reply:
How about "Providing empty goals means to use all the ones declared in the default.goals cruise control configuration parameter" (to make it clear where that default.goals is)?

Reply:
I am fine with it.